salt/etc-hosts/update-pre-reboot.sls

```diff
@@ -1,8 +1,16 @@
 # Remove our /etc/hosts entries, so that Systemd/Wicked hostname
 # logic doesn't pick up out /etc/hosts entry names.
```

Comment:
So - while this isn't a new issue, it's worth pointing out this won't run when an operator just reboots a machine, e.g. with `shutdown -r now` or due to power loss etc. What exactly happens when this code doesn't run and the reboot happens?

Reply:
I don't fully understand you question. It will happen the same that happened before when `/etc/hosts` was managed by a Salt state.

Reply:
@kiall I agree with @inercia here - this was a problem before, and I doubt that we can fix it for all cases. In the end there could always be a power outage that could trigger not running our code.